src/cdk/a11y/focus-monitor/focus-monitor.ts

```diff
@@ -110,6 +111,14 @@ export class FocusMonitor implements OnDestroy {
   /** The number of elements currently being monitored. */
   private _monitoredElementCount = 0;
 
+  /**
+   * Keeps track of the root nodes to which we've currently bound a focus/blur handler,
+   * as well as the number of monitored elements that they contain. We have to treat focus/blur
+   * handlers differently from the rest of the events, because the browser won't emit events
+   * to the document when focus moves inside of a shadow root.
+   */
+  private _rootNodeFocusListenerCount = new Map<HTMLElement|Document, number>();
```

Comment:
To be safe, we should scan through this in `ngOnDestroy` and unlisten to any remaining entries

Reply:
We're looping through and calling `stopMonitoring` on all the remaining elements which eventually empties this out as well.